docs/getting_started.rst

```diff
@@ -3,21 +3,113 @@ Getting Started
 
 **Table of Contents**
 
-* :ref:`Installation`
+* :ref:`Installation and deployment`
 
 * :ref:`Example`
 
+* :ref:`Running SOSS`
+
 * :ref:`Getting Help`
 
-Installation
-^^^^^^^^^^^^
 
-To build `soss` it is recommended use a
-`colcon workspace <https://colcon.readthedocs.io/en/released/user/quick-start.html>`__.
-The `soss` repo consists of many cmake packages which can be configured and built manually,
-but colcon makes the job much smoother.
+Installation and deployment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this section we skecth the steps necessary for installing SOSS and running a SOSS instance.
+
+Note that the workflow is
+dependent on the specific systems that are being communicated, given that each is brougth into the SOSS world via a dedicated System Handle (SH).
+An example of how these generic steps are declined in a concrete use-case can be found in the following section.
+
+As a first step, you will need to create a `colcon workspace <https://colcon.readthedocs.io/en/released/user/quick-start.html>`__
+to clone the SOSS repo.
+This repo contains the soss core library and the SHs for some of the protocols that are integrated into the SOSS world.
+It consists of many cmake packages which can be configured and built manually, but we recommend to use
+a `colcon workspace <https://colcon.readthedocs.io/en/released/user/quick-start.html>`__, which makes the job much smoother.
+To do so, create a workspace/soss folder and clone SOSS into it:
+
+.. code-block:: bash
+
+    cd
+    mkdir -p workspaces/soss
+    cd workspaces/soss
+    git clone ssh://git@github.com/eProsima/soss_v2 src/soss --recursive -b feature/xtypes-dds
+
+Note: the `--recursive` flag is mandatory to download some required third-parties.
+
+The second step consists in building the required packages. For this, you will need to run `colcon build`:
+
+.. code-block:: bash
+
+    colcon build
+
+with the possible addition of flags depending on the specific use-case. Notice also that in most situations you will have to execute some additional commands before running the `colcon build`, e. g. sourcing an external overlay,
+as is the case for ROS2 and ROS1 (see e. g. next section).
+
+Once that's finished building, you can source the new colcon overlay:
+
+.. code-block:: bash
+
+    source install/setup.bash
+
+Now, if the SHs of the systems to be communicated are all hosted by the SOSS repo, you are ready to run a SOSS instance. 
+If, on the other hand, the SHs of the middlewares involved are not built-in, you need to clone their specific repos into
+dedicated workspaces (soss-mw, where mw indicates the middleware) located in the workspace/soss folder.
```

Comment:
I changed all this. However, I'm not really satisfied because the example below is not a nice and elegant reflection of this generic part, but presents additional steps instead. I think it made sense to make the two approaches as similar as possible, the generic one and the example.

Reply:
Review this when you are satisfied :).

Reply:
Still thinking about it.